client/state/ui/reducer.js

```diff
@@ -74,6 +75,13 @@ export const isPreviewShowing = createReducer( false, {
 		isShowing !== undefined ? isShowing : state,
 } );
 
+// FIXME: We can't detect window size on the server, so until we have more intelligent detection,
+// use 769, which is just above the general maximum mobile screen width.
+export const viewportWidth = createReducer( 769, {
+	[ VIEWPORT_WIDTH_SET ]: ( state, { width } ) =>
+		! isNaN( width ) ? width : state,
+} );
```

Comment:
I quite like your first reducer suggestion 😄; I do that sort of dict-switching often but I don't think I've done/seen it in reducers yet. I might use that next time.

> Just curious, what do you find more expressive about it?

I think our debating hinges on details. :) I think that, same way you describe `createReducer` as adding _a lot of syntax noise_, that's how I see the typical alternative, which is `switch` blocks: noise with `case`, `return`, plus the error-prone requirement to avoid falling through to the next case, or defaulting to returning current state.

> It's confusing to me because it does stuff behind the scenes that isn't all that clear whereas when we have really simple reducers then what-you-see-is-what you get, plus it's that many fewer imports, that much less module bloat, that many fewer function calls and memory allocations…

I agree with all of that, no counterarguments, except maybe for:

> because it does stuff behind the scenes that isn't all that clear

This could be said of several instrumental pieces of Calypso, from implicit code splitting to decoupling data consumption from data fetching (e.g. use of `Query*` components as a statement of data need). There's always a balance to strike with abstractions — we'd like them to be few, sounds and broad-reaching — but one like `createReducer` is IMO worth learning (it's not _that_ magic, after all), since it affects such a pivotal aspect of state management.